src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,366 @@
+     * @return void
+     */
+    protected function updateCompanyUserFieldChoices(FormInterface $form, int $idParentCompanyBusinessUnit): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER, SelectType::class, array_merge(
+            $this->getCompanyUserFieldOptions(),
+            [
+                'choices' => $this->getCompanyUserChoices($idParentCompanyBusinessUnit),
+            ]
+        ));
+    }
+
+    /**
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return array
+     */
+    protected function getCompanyBusinessUnitChoices(int $idParentCompanyBusinessUnit): array
+    {
+        $companyBusinessUnitCollectionTransfer = $this->getFactory()
+            ->getCompanyBusinessUnitFacade()
+            ->getCompanyBusinessUnitCollection(
+                (new CompanyBusinessUnitCriteriaFilterTransfer())
+                    ->setIdParentCompanyBusinessUnit($idParentCompanyBusinessUnit)
+            );
+
+        $parentCompanyBusinessUnitTransfer = $this->getFactory()
+            ->getCompanyBusinessUnitFacade()
+            ->findCompanyBusinessUnitById($idParentCompanyBusinessUnit);
+
+        if (!$parentCompanyBusinessUnitTransfer) {
+            return [];
+        }
+
+        $companyBusinessUnits = [];
+        $companyBusinessUnitTransfers = $companyBusinessUnitCollectionTransfer->getCompanyBusinessUnits()->count()
+            ? $companyBusinessUnitCollectionTransfer->getCompanyBusinessUnits()
+            : [$parentCompanyBusinessUnitTransfer];
+
+        foreach ($companyBusinessUnitTransfers as $companyBusinessUnitTransfer) {
+            $companyBusinessUnits[$companyBusinessUnitTransfer->getName()] = $companyBusinessUnitTransfer->getIdCompanyBusinessUnit();
+        }
+
+        return $companyBusinessUnits;
+    }
+
+    /**
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return array
+     */
+    protected function getCompanyUserChoices(int $idParentCompanyBusinessUnit): array
+    {
+        $companyUserCollectionTransfer = $this->getFactory()
+            ->getCompanyUserFacade()
+            ->getCompanyUserCollection(
+                (new CompanyUserCriteriaFilterTransfer())
+                    ->setIdCompanyBusinessUnit($idParentCompanyBusinessUnit)
+            );
+
+        $companyUsers = [];
+
+        foreach ($companyUserCollectionTransfer->getCompanyUsers() as $companyUserTransfer) {
+            $customerTransfer = $companyUserTransfer->getCustomer();
+            $choiceLabel = sprintf(
+                '%s %s (%s)',
+                $customerTransfer->getFirstName(),
+                $customerTransfer->getLastName(),
+                $customerTransfer->getEmail()
+            );
+
+            $companyUsers[$choiceLabel] = $companyUserTransfer->getIdCompanyUser();
+        }
+
+        return $companyUsers;
+    }
+
+    /**
+     * @return array
+     */
+    protected function getCompanyBusinessUnitFieldOptions(): array
```

Comment:
useless method

Reply:
IMO it's not useless, it reduces code duplication, it used 3 times.

Does it make sense? 

It could be named `getDefaultCompanyBusinessUnitFieldOptions`, and moved to `DataProvider` if you wish.

Reply:
Discussed, refactored